Source/init.cpp

```diff
@@ -118,7 +116,7 @@ HANDLE init_test_access(char *mpq_path, char *mpq_name, char *reg_loc, int flags
 
 	for (int i = 0; i < 2; i++) {
 		snprintf(mpq_path, MAX_PATH, "%s%s", Buffer[i], mpq_name);
-		if (SFileOpenArchive(mpq_path, 0, flags, &archive)) {
+		if (SFileOpenArchive(mpq_path, 0, FS_PC, &archive)) {

```

Comment:
@MBeijer 
Interesting `FS_PC` is `0` meaning that StormLib will default to read/write.

Could you test the following:
```suggestion
		if (SFileOpenArchive(mpq_path, flags, MPQ_FLAG_READ_ONLY, &archive)) {
```
or
```suggestion
		if (SFileOpenArchive(mpq_path, 0, MPQ_FLAG_READ_ONLY, &archive)) {
```
I would say this should be fixed in StormLib (even if it just does the same for `r` vs `rw`) rather then setting to `rw`.

Reply:
This works.